backend/src/roles/roles.guard.ts

```diff
@@ -0,0 +1,45 @@
+import { Injectable, CanActivate, ExecutionContext } from '@nestjs/common';
+import { Reflector } from '@nestjs/core';
+import { Roles } from './roles.enum';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Junior } from '../junior/junior.entity';
+import { Repository } from 'typeorm';
+import { Admin } from '../admin/admin.entity';
+@Injectable()
+export class RolesGuard implements CanActivate {
+    constructor(
+        private readonly reflector: Reflector,
+        @InjectRepository(Junior)
+        private readonly juniorRepo: Repository<Junior>,
+        @InjectRepository(Admin)
+        private readonly adminRepo: Repository<Admin>) { }
+
+    async canActivate(context: ExecutionContext): Promise<boolean> {
+        const roles = this.reflector.get<Roles[]>('roles', context.getHandler());
+        if (!roles) {
+            return true;
+        }
+        const request = context.switchToHttp().getRequest().user;
+        const id = request.userId;
+        const identity = request.user;
+        const userRoles = await this.getUserRoles(id, identity);
+        const hasRole = () => userRoles.some((role) => roles.includes(role));
+        return id && identity && hasRole();
+    }
+
+    private async getUserRoles(id: string, identity: string): Promise<Roles[]> {
```

Comment:
Should we populate the request with users roles somewhere in auth pipeline?

EDIT: I guess it doesnt really matter where we do it as we dont need the data later anyways. (Or do we?)

Reply:
As  discussed
We currently do not have a need for this, it would be better to add the feature when needed rather than introducing unused code.